gui/system/models.py

```diff
@@ -917,9 +917,26 @@ def cert_ncertificates(self):
     def cert_DN(self):
         self.__load_thingy()
 
+        obj = self.__get_thingy()
         parts = []
-        for c in self.__get_thingy().get_subject().get_components():
+        for c in obj.get_subject().get_components():
             parts.append("%s=%s" % (c[0].decode(), c[1].decode('utf8')))
+
+        if not any(
+            'subjectAltName' == c for c in map(
+                lambda t: t[0].decode(),
+                obj.get_subject().get_components()
+            )
+        ):
```

Comment:
Does it even make sense do this first iteration if we have to iterate all over again if it evaluates to true? Or is this a performance improvement of some sort?

Reply:
We can use a boolean flag instead, this seemed more neat and the loop was small so went ahead with it. I'll have it updated ASAP